File: C#/EntityFramework_ORM/DAL/Repositories/EmployeeRepository.cs
using DAL.Repositories;
using DAL.Entities;
using DAL.Data;
using Microsoft.EntityFrameworkCore;

namespace DAL.Repositories;

public class EmployeeRepository /*: GenericRepository<Employee, int>*/
{
    private MyAppDbContext _context;
    public EmployeeRepository(MyAppDbContext context)
    {
        _context = context;
    }
    public async Task<List<Employee>> GetAllAsync() => await _context.Set<Employee>().ToListAsync();

    public async Task<Employee?> GetByIdAsync(int id) => await _context.Set<Employee>().FindAsync(id);

    public async Task<Employee> AddAsync(Employee entity)
    {
        await _context.Set<Employee>().AddAsync(entity);
        await _context.SaveChangesAsync();
        return entity;
    }

    public async Task UpdateAsync(Employee entity)
    {
        _context.Set<Employee>().Update(entity);
        await _context.SaveChangesAsync();
    }

    public async Task DeleteAsync(int id)
    {
        var entity = await GetByIdAsync(id);
        if (entity != null)
        {
            _context.Set<Employee>().Remove(entity);
            await _context.SaveChangesAsync();
        }
    }
}

File: Go/GORM_ORM/dal/repositories/employeeRepository.go
package repositories

import (
	"GORM_ORM/dal/entities"

	"gorm.io/gorm"
)

type EmployeeRepository interface {
	GetAll() ([]entities.Employee, error)
	GetByID(id int) (*entities.Employee, error)
	Create(entity *entities.Employee) error
	Update(entity *entities.Employee) error
	Delete(id int) error
}

type employeeRepository struct {
	genericRepo GenericRepository[entities.Employee]
}

func NewEmployeeRepository(db *gorm.DB) EmployeeRepository {
	return &employeeRepository{
		genericRepo: NewGenericRepository[entities.Employee](db),
	}
}

func (r *employeeRepository) GetAll() ([]entities.Employee, error) {
	return r.genericRepo.GetAll()
}

func (r *employeeRepository) GetByID(id int) (*entities.Employee, error) {
	return r.genericRepo.GetByID(id)
}

func (r *employeeRepository) Create(entity *entities.Employee) error {
	return r.genericRepo.Create(entity)
}

func (r *employeeRepository) Update(entity *entities.Employee) error {
	return r.genericRepo.Update(entity)
}

func (r *employeeRepository) Delete(id int) error {
	return r.genericRepo.Delete(id)
}
